astropy/io/fits/column.py

```diff
@@ -1389,19 +1389,21 @@ def __getattr__(self, name):
 
         Implements for example self.units, self.formats, etc.
         """
-
         cname = name[:-1]
         if cname in KEYWORD_ATTRIBUTES and name[-1] == 's':
             attr = []
-            for col in self:
+            for col in self.columns:
                 val = getattr(col, cname)
-                if val is not None:
-                    attr.append(val)
-                else:
-                    attr.append('')
+                attr.append(val if val is not None else '')
```

Comment:
@astrofrog - I think so, but the original code uses `is not None`. I'm not sure if there are case where `val` could be a falsy value that should not be translated to a empty string. So, in doubt, better stick with the original implementation.